flink-connectors/flink-connector-jdbc/src/test/java/org/apache/flink/connector/jdbc/xa/JdbcExactlyOnceSinkE2eTest.java

```diff
@@ -102,19 +212,108 @@ public void cancel() {
 
         @Override
         public void notifyCheckpointComplete(long checkpointId) {
-            if (checkpointId == this.checkpointAfterData) {
-                dataCheckpointed = true;
+            if (checkpointId == this.lastCheckpointId) {
+                lastSnapshotConfirmed = true;
+            }
+        }
```

Comment:
would this cause test in-stablity?

Reply:
I don't see any issue here, could you explain your concern?